src/platform/startup/react.js

```diff
@@ -39,8 +39,13 @@ export default function startReactApp(component, root = null) {
     mountElement = document.getElementById('react-root');
   }
   // eslint-disable-next-line scanjs-rules/call_addEventListener
-  document.addEventListener('DOMContentLoaded', () => {
+
+  if (document.readyState !== 'loading') {
```

Comment:
at this point proxied requests have already triggered the DOMContentLoaded event, and we've added new elements to the page. We're doing a naive check here to see if we can just render the component immediately.

Reply:
I wonder if having a second optional parameter for each startup function (startMegaMenuWidget, etc.) to override `startReactApp` would makes things any easier, or just more work -

```jsx
export default function startSomeComponent(store, callback = startReactApp) {
    callback((
      <Provider store={store}>
        <Announcement/>
      </Provider>
    ), elementId);
}
```